tutorials/ios-chatbot/index.md

```diff
@@ -0,0 +1,145 @@
+---
+title: How to build a conversational app using Cloud Machine Learning APIs (Part 1 of 3)
+description: How to build a conversational app using Cloud Machine Learning APIs (Part 1 of 3).
+author: PokerChang
+tags: Cloud Functions, Dialogflow, API.AI, Webhooks, Localization, Chatbot, Machine Learning API, Transation, Vision, Speech
+date_published: 2017-01-02
+---
+
+For consumers, conversational apps (such as chatbot) are among the most visible examples of machine learning in action. For developers, building a conversational app is instructive for understanding the value that machine-learning APIs bring to the process of creating completely new user experiences.
+
+In this three-part post, we will show you how to build an example "[tour guide](https://youtu.be/qDAP3ZFjO48)" app for Apple iOS that can see, listen, talk and translate via [API.AI](https://api.ai/) (a developer platform for creating conversational experiences) and [Google Cloud ML APIs for Speech, Vision and Translate](https://cloud.google.com/products/machine-learning/). You will also see how easy it is to support multiple languages on these platforms. 
+
+[![English Demo](http://img.youtube.com/vi/qDAP3ZFjO48/0.jpg)](https://youtu.be/qDAP3ZFjO48)
+
+The three parts will focus on the following topics:
+
+Part 1
+
+
+
+*   Overview
+*   Architecture
+*   API.AI intents
+*   API.AI contexts
+
+[Part 2]
+
+*   API.AI webhook with [Cloud Functions](https://cloud.google.com/functions/)
+*   [Cloud Vision API](https://cloud.google.com/vision/)
+*   [Cloud Speech API](https://cloud.google.com/speech/)
+*   [Cloud Translation API](https://cloud.google.com/translate/)
+*   Support multiple languages
+
+[Part 3]
+
+*   Support the Google Assistant via Actions on Google Integration
+
+This post is Part 1 and Part 2 and 3 will be published in the following weeks.
+
+
+## Architecture
+
+![alt_text](chatbots-8.png "Architecture")
+
+## Using API.AI
+
+
+API.AI is a platform for building natural and rich conversational experiences. For our example, it will handle all core conversation flows in the tour guide app. (Note that API.AI provides great [documentation and a sample app](https://github.com/api-ai/apiai-ios-client) for its iOS SDK. SDKs for [other platforms](https://docs.api.ai/docs/sdks) are also available, so you could easily extend this tour guide app to support Android.)
+
+
+### Create Agent
+
+The first step is to create a "Tour Guide Agent." 
+
+
+### Create Intents
+
+![alt_text](chatbots-3.png "Create Intents Screenshot")
+
+To engage users in a conversation, we first need to understand what users are saying to the agent and we do that with intents and entities. Intents map what your users say to what your conversational experience should do. Entities are used to extract parameter values from use queries.
```

Comment:
In first sentence, we need a comma between **agent** and **and**.

Reply:
In last sentence, should that say **user queries** instead of **use queries**?

Reply:
Suggested change for the last sentence:
Entities extract parameter values from use (or user) queries.